test/hotspot/jtreg/compiler/valhalla/inlinetypes/TestArrays.java

```diff
@@ -3773,4 +3773,16 @@ public double test154() {
     public void test154_verifier() {
         Asserts.assertEquals(test154(), rD);
     }
+
+    @Test
+    @IR(applyIf = {"UseArrayLoadStoreProfile", "true"},
```

Comment:
Maybe add a small comment here explaining what the test / IR rule is supposed to catch.